internal/kafka/internal/handlers/cluster.go

```diff
@@ -132,3 +134,38 @@ func (h clusterHandler) List(w http.ResponseWriter, r *http.Request) {
 
 	handlers.HandleList(w, r, cfg)
 }
+
+func (h clusterHandler) DeregisterEnterpriseCluster(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	clusterID := mux.Vars(r)["id"]
+	forceDeleteFlag := r.URL.Query().Get("force")
+
+	if forceDeleteFlag == "" {
+		forceDeleteFlag = "false"
+	}
+
+	forceBoolValue, err := strconv.ParseBool(forceDeleteFlag)
+
+	validateForceParam := func() handlers.Validate {
+		return func() *errors.ServiceError {
+			if err != nil {
+				return errors.GeneralError("failed to parse query param force: %s", forceDeleteFlag)
+			}
+			return nil
+		}
+	}
+
+	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			handlers.ValidateAsyncEnabled(r, "deleting enterprise cluster"),
+			validateForceParam(),
+			ValidateKafkaClaims(ctx, ValidateOrganisationId()),
+			validateEnterpriseClusterEligibleForDeregistration(ctx, clusterID, forceBoolValue, h.clusterService),
+		},
+		Action: func() (i interface{}, serviceError *errors.ServiceError) {
+			// cluster deletion from db and cleanup will be handled in a follow up Jira
```

Comment:
note that this will be done as a follow up. in this PR no db entry deletion takes place and its deliberate

Reply:
FYI @pb82

Reply:
what is the difference between a hard- and a soft delete in the database? Do we only set a deleted flag?